deepspeed/runtime/zero/stage3.py

```diff
@@ -2845,3 +2848,17 @@ def _handle_overflow(cpu_sum, x, i):
         logger.info(
             f"rank {rank} detected overflow {cpu_sum} in tensor {i}:{t_i} shape {x.shape}"
         )
+
+
+def remove_module_hooks(module, hook_location='ds_hook_handles', recurse=True):
+    """
+    Function to remove all module forward/backward hooks. This is useful in cases
+    where the model may be wrapped more than once and the module hooks should
+    be removed first.
+    """
+    if hasattr(module, hook_location):
+        for k, v in getattr(module, hook_location).items():
+            v.remove()

```

Comment:
@SeanNaren We should also remove the hook handle from the dictionary after removing the hook itself no?

Reply:
Good catch! thanks @samyam :) I'll make the change. 

Haven't had time to dig into what this actually fixes on the Lightning side, but will get to it as soon as I can, not a high priority issue as the fix exists upstream in lightning for now as well!